sisl/geometry.py

```diff
@@ -1019,27 +1021,451 @@ def overlap(self, other, eps=0.1, offset=(0., 0., 0.), offset_other=(0., 0., 0.)
+                # of the missing unique items
+                try:
+                    none_idx = groups.index(None)
+
+                    # we have a None (ensure we use a list, tuples are
+                    # immutable)
+                    groups = list(groups)
+                    groups[none_idx] = []
+
+                    uniq_groups = np.unique(concatenate(groups))
+                    # add a new group that is in uniq_vals, but not in uniq_groups
+                    rest = uniq_vals[isin(uniq_vals, uniq_groups, invert=True)]
+                    groups[none_idx] = rest
+                except ValueError:
+                    # there is no None in the list
+                    pass
+
+            for at in atom:
+                # reduce search
+                at_vals = vals[at]
+                # loop group values
+                for group in groups:
+                    # retain original indexing
+                    nl.append(at[isin(at_vals, group)])
+            return nl
+
+        def _group(method_group, atom, **kwargs):
+            """
+            Group based sorting is based on a named identification.
+
+            group: str or tuple of (str, list of lists)
+
+            symbol: order by symbol (most cases same as Z)
+            Z: order by atomic number
+            tag: order by atom tag (should be the same as specie)
+            specie/species: order by specie (in order of contained in the Geometry)
+            """
+            # Create new list
+            nl = NestedList()
+            if isinstance(method_group, str):
+                method = method_group
+                groups = []
+            else:
+                method, *in_groups = method_group
+
+                # Ensure all groups are lists
+                groups = []
+                NoneType = type(None)
+                for group in in_groups:
+                    if isinstance(group, (tuple, list, ndarray, NoneType)):
+                        groups.append(group)
+                    else:
+                        groups.append([group])
+
+            method = method.lower()
+
+            if method == 'z':
+                return _group_vals(self.atoms.Z, groups, atom, **kwargs)
+            elif method in ['specie', 'species']:
+                return _group_vals(self.atoms.specie, groups, atom, **kwargs)
+            elif method == 'tag':
+                # create numpy array
+                strs = np.array(list(map(lambda a: a.tag, self.atoms)))
+                return _group_vals(strs, groups, atom, **kwargs)
+            elif method == 'symbol':
+                strs = np.array(list(map(lambda a: a.symbol, self.atoms)))
+                return _group_vals(strs, groups, atom, **kwargs)
+            else:
+                raise ValueError(f"{self.__class__.__name__}.sort group only supports ['Z','species','tag','symbol'] for sorting")
+        funcs["group"] = _group
+
```

Comment:
Not sure why, but the command `sgeom in.xyz --sort "group=Z" out.xyz` fails for me with a `ValueError: invalid literal for int() with base 10: 'Z'`

Reply:
Thanks, now I fixed it :)